pkg/kubelet/kubelet_test.go

```diff
@@ -455,15 +457,7 @@ func TestSyncPodsWithTerminationLog(t *testing.T) {
 
 // drainWorkers waits until all workers are done.  Should only used for testing.
 func (kl *Kubelet) drainWorkers() {
-	for {
-		kl.podWorkers.lock.Lock()
-		length := len(kl.podWorkers.workers)
-		kl.podWorkers.lock.Unlock()
-		if length == 0 {
-			return
-		}
-		time.Sleep(time.Millisecond * 100)
-	}
+	kl.podWorkers.waitGroup.Wait()
```

Comment:
WDYT if instead of this as part of newTestKubelet we create a podWorkers where the syncPodFun is:

```Go
func() {
  syncPod()
  waitgroup.Done()
}
```

Just trying to find a way to keep the test-only code outside of the production code.